docs/creating_component_code_review.md

```diff
@@ -11,6 +11,8 @@ A checklist of things to do when you're adding a new component.
  1. Follow our [Style guidelines](development_guidelines.md)
  2. Use existing constants from [`const.py`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/const.py)
     * Only add new constants to `const.py` if they are widely used. Otherwise keep them on components level
+ 3. Add yourself to [`CODEOWNERS`](https://github.com/home-assistant/home-assistant/blob/dev/CODEOWNERS) so you [automatically get requested to review](https://help.github.com/en/articles/about-code-owners#codeowners-syntax) the code when it changes in the future
+ 4. Rebase (**do not merge**) to (catch up with latest dev branch)[https://developers.home-assistant.io/docs/en/development_catching_up.html]
```

Comment:
Yes!!

Reply:
Why? A merge ends up the same since we squash. However, rebase is much harder to work with for the contributor. Every week has several rebase-gone-bad submissions notifying everyone and rebase conflicts are some of the hardest situations to help people with in chat.

Note, I am not arguing against rebase as such but if the contributor knows how to rebase she will do it anyway.

Reply:
Rebase should be a requirement.  While what you said is true when the PR is squashed, merging creates unrelated changes (add/subtractions/commits) in the PR itself and makes it significantly harder to review and follow-up on requested changes.  If there are conflicts, it's the roughly the same difficulty to resolve regardless of the flow used.

Reply:
Huh? The opposite seems to be true. After a rebase and force-push, one must restart the review because you have no idea what changed.

My point about difficulty is that a rebase conflict can propagate through all commits on the branch but with a merge, it must only be resolved once. And you get a record of _how_ it was resolved.

_If you know what you are doing_, rebase can give you a cleaner PR. If not, it can make you lose your work. I don't think we should make blanket statements when we don't know which category the contributor is in.

Reply:
Generally I try to avoid having contributors catch up with latest dev. It's rarely needed. If they have to, they should rebase, but as @amelchio said, it often goes wrong because a push will fail, and git instructs them to merge the changes from upstream down 🤷‍♂️